### PR TITLE
Update example code: setupOffscreenDocument

### PR DESCRIPTION
**Pull request by hanguokai**
Replace hasOffscreenDocument() with setupOffscreenDocument() to resolve concurrency issues

### CODE REVIEW

**Comment by hanguokai** [comment]
@sebastianbenz [suggested me](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/D5Jg2ukyvUc/m/JFhAYDEpAAAJ) to create this PR.

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 1]
```diff
@@ -40,44 +40,54 @@ Use `chrome.offscreen.createDocument()` and `chrome.offscreen.closeDocument()` f
...
+async function setupOffscreenDocument(path) {
   // Check all windows controlled by the service worker to see if one 
   // of them is the offscreen document with the given path
+  let exist = false;
   const offscreenUrl = chrome.runtime.getURL(path);
   const matchedClients = await clients.matchAll();
   for (const client of matchedClients) {
     if (client.url === offscreenUrl) {
-      return true;
+      exist = true;
```
Comment (sebastianbenz) [inline site/en/docs/extensions/reference/offscreen/index.md:63]: I'd prefer an early exit here. This also avoid the additional variable + if below. .

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 2]
```diff
@@ -40,44 +40,54 @@ Use `chrome.offscreen.createDocument()` and `chrome.offscreen.closeDocument()` f
...
+      break;
+    }
+  }
+  
+  // create offscreen document if it's not existing
+  if (!exist) {
+    if (creating) {
+      await creating;
```
Comment (sebastianbenz) [inline site/en/docs/extensions/reference/offscreen/index.md:71]: Do we need to `await` here? We could also simply return `creating` and avoid the `else` block below.

**Review by sebastianbenz — COMMENTED** [review]
Amazing! Thanks a lot. Left a suggestion to use early exit otherwise LGTM.

File: site/en/docs/extensions/reference/offscreen/index.md (hunk 1 above)
Comment (hanguokai) [reply site/en/docs/extensions/reference/offscreen/index.md]: Thanks, early exit here more simple

File: site/en/docs/extensions/reference/offscreen/index.md (hunk 2 above)
Comment (hanguokai) [reply site/en/docs/extensions/reference/offscreen/index.md]: I prefer the current way here, which looks more symmetrical

**Comment by hanguokai** [comment]
Thanks for suggestion, I submitted a new version, that is a bit simpler

**Review by sebastianbenz — APPROVED** [review]
Thanks a lot - looks great!

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 3]
```diff
@@ -40,44 +40,50 @@ Use `chrome.offscreen.createDocument()` and `chrome.offscreen.closeDocument()` f
...
+  url: 'off_screen.html',
   reasons: ['CLIPBOARD'],
   justification: 'reason for needing the document',
 });
 
 chrome.offscreen.closeDocument()
 ```
 
-The following example shows how to check for existing offscreen documents. The `hasOffscreenDocument()` function uses [clients.matchAll()](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find existing offscreen documents. Please note: the function assumes that your extension uses a single offscreen document. 
+The following example shows how to ensure that offscreen documents have already been created. The `setupOffscreenDocument()` function uses [clients.matchAll()](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find existing offscreen documents. If no existing offscreen documents, create it. Please note: the function assumes that your extension uses a single offscreen document.
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/offscreen/index.md:51]:
```suggestion
The following example shows how to ensure that an offscreen document has already been created. The `setupOffscreenDocument()` function calls [`clients.matchAll()`](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find the existing offscreen document. If no offscreen document exists, is created. Note that a service worker can only have one offscreen document. 
```

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 4]
```diff
@@ -40,44 +40,50 @@ Use `chrome.offscreen.createDocument()` and `chrome.offscreen.closeDocument()` f
...
+      justification: 'reason for needing the document',
+    });
+    await creating;
+    creating = null;
+  }
 }
 ```
 
-Before trying to create a new offscreen document, call `hasOffscreenDocument()` to make sure that there is no existing offscreen document, as demonstrated in the following example. 
+Before sending a message to an offscreen document, call `setupOffscreenDocument()` to make sure that there is existing offscreen document, as demonstrated in the following example.
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/offscreen/index.md:81]:
```suggestion
Before sending a message to an offscreen document, call `setupOffscreenDocument()` to make sure that there is an existing offscreen document, as demonstrated in the following example. 
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: site/en/docs/extensions/reference/offscreen/index.md (hunk 3 above)
Comment (jpmedley) [reply site/en/docs/extensions/reference/offscreen/index.md:51]: I fixed the first item.

I didn't remove the 'Please note...'. I just made it a bit pithier.

Comment (jpmedley) [reply site/en/docs/extensions/reference/offscreen/index.md:51]: I appreciate your time and input, but this is problematic. I think you're trying to future proof this, which is a good instinct. But, baseline documentation shouldn't make promises about future product work. If we've done that elsewhere, I need to fix it.

What this line needs to do is say in a straightforward way that the function gets the existing offscreen document or creates it if it doesn't already exist and that it does so using...

In other words, stick to the facts.

**Comment by hanguokai** [comment]
I submitted a new version followed by Joe's suggestion. Please review. Thanks!

**Review by jpmedley — APPROVED** [review]
